src/site/content/en/blog/nfc/index.md

```diff
@@ -0,0 +1,911 @@
+mechanism works for writing NFC tags as it uses the same permission under the
+hood.
+
+```js
+const reader = new NDEFReader();
+
+async function startScanning() {
+  await reader.scan();
+  reader.onreading = event => {
+    /* handle NDEF messages */
+  };
+}
+
+const nfcPermissionStatus = await navigator.permissions.query({ name: "nfc" });
+if (permissionStatus.state === "granted") {
+  // NFC access was previously granted, so we can start NFC scanning now.
+  startScanning();
+} else {
+  // Show a "scan" button.
+  document.querySelector("#scanButton").style.display = "block";
+  document.querySelector("#scanButton").onclick = event => {
+    // Prompt user to allow UA to send and receive info when they tap NFC devices.
+    startScanning();
+  };
+}
+```
+
+### Abort NFC operations
+
+Using the [AbortController] primitive makes it easy to abort NFC operations and
+other operations that use it. The example below shows you how to pass the
+`signal` of an AbortController through the options of NDEFReader `scan()` and
+NDEFWriter `write()` methods and abort both NFC operations at the same time.
+
+```js
+const abortController = new AbortController();
+abortController.signal.onabort = event => {
+  // All NFC operations have been aborted.
+};
+
+const reader = new NDEFReader();
+await reader.scan({ signal: abortController.signal });
+
+const writer = new NDEFWriter();
+await writer.push("Hello world", { signal: abortController.signal });
+
+document.querySelector("#abortButton").onclick = event => {
+  abortController.abort();
+};
+```
+
+### Read and write a text record
+
+The text record `data` can be decoded with a `TextDecoder` instantiated with the
+record `encoding` property. Note that the language of the text record is
+available through its `lang` property.
+
+```js
+function readTextRecord(record) {
+  console.assert(record.recordType === "text");
+  const textDecoder = new TextDecoder(record.encoding);
+  console.log(`Text: ${textDecoder.decode(record.data)} (${record.lang})`);
+}
+```
+
+Writing a simple text record can be done by passing a string to the NDEFWriter
+`write()` method.
+
+```js
+const writer = new NDEFWriter();
+await writer.write("Hello World");
+```
+
+Text records are UTF-8 by default and assume the current document's language but
+both properties (`encoding` and `lang`) can be specified using the full syntax
+for creating a custom NDEF record.
+
+```js
+function a2utf16(string) {
+  let result = new Uint16Array(string.length);
+  for (let i = 0; i < string.length; i++) {
+    result[i] = string.codePointAt(i);
+  }
+  return result;
+}
+
+const textRecord = {
+  recordType: "text",
+  lang: "fr",
+  encoding: "utf-16",
+  data: a2utf16("Bonjour, François !")
+};
+
+const writer = new NDEFWriter();
+await writer.write({ records: [textRecord] });
+```
+
+### Read and write a URL record
+
+The URL record's `data` can be decoded with a `TextDecoder`.
```

Comment:
Use `TextDecoder` to decode the record's `data`.

Reply:
Done